src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -325,7 +325,6 @@ public async ValueTask<bool> WaitFor100ContinueAsync(CancellationToken cancellat
 
             private void SendReset()
             {
-                Debug.Assert(!Monitor.IsEntered(SyncObject));
```

Comment:
I think we should probably be asserting the opposite now: that is, that we have the lock. Right?

This probably means other callers need to be changed to hold the lock as well -- but I think this is necessary to ensure they don't have races as well.

Reply:
Assert for Monitor.IsEntered is added.